sqlglot/expressions.py

```diff
@@ -2603,10 +2603,32 @@ class DistKeyProperty(Property):
     arg_types = {"this": True}
 
 
+# [starrocks distributed Doc](https://docs.starrocks.io/docs/sql-reference/sql-statements/data-definition/CREATE_TABLE/#distribution_desc)
+# [doris distributed Doc](https://doris.apache.org/docs/sql-manual/sql-statements/Data-Definition-Statements/Create/CREATE-TABLE?_highlight=create&_highlight=table#distribution_desc)
+class DistributedByHashProperty(Property):
+    arg_types = {"expressions": True, "buckets": False, "sorted_by": False}
+
+    @property
+    def auto_bucket(self) -> bool:
+        return self.args.get("buckets") is None
+
+
+class DistributedByRandomProperty(Property):
+    arg_types = {"expressions": True, "buckets": False, "sorted_by": False}
```

Comment:
Can't we have an "order" arg instead of `"sorted_by"` and then use `_parse_order` to capture the syntax? With this approach we're duplicating the generation for `ORDER BY`.